src/substrate/types.ts

```diff
@@ -126,12 +127,29 @@ export enum EventKind {
   JudgementGiven = 'identity-judgement-given',
   IdentityCleared = 'identity-cleared',
   IdentityKilled = 'identity-killed',
+
+  HeartbeatReceived = 'heartbeat-received',
+  SomeOffline = 'some-offline'
 }
 
 interface IEvent {
   kind: EventKind;
 }
 
+/**
+ * ImOnline Events
+ */
+export interface IHeartbeatReceived extends IEvent {
+  kind:EventKind.HeartbeatReceived;
+  authorityId: AuthorityId;
+}
+
+export interface ISomeOffline extends IEvent {
+  kind:EventKind.SomeOffline;
+  sessionIndex: number;
+  validators: Array<IdentificationTuple>
```

Comment:
This is another case where we don't want to use the substrate type directly, because we can't marshal `IdentitificationTuple`. Please unwrap it to contain the "raw" types, maybe in a tuple literal.

Reply:
Would a `toHuman()` representation work ok here @jnaviask ?

Reply:
It would technically work but we wouldn't have any typings for the fields produced (i.e. it would still be an "any" or "object" type, and not have any data about what to expect).